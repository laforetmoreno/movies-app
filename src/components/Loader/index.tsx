import React from "react";

import style from "./style.module.scss";

interface Props {
  size?: number;
  weight?: number;
  color?: string;
  minHeight?: boolean;
}

const Loader = ({ size = 80, weight = 6, color = "#f90", minHeight }: Props) => {
  return (
    <div className={`${style.ldsRing} ${minHeight && style.minHeight}`}>
      <div
        style={{
          width: `${size}px`,
          height: `${size}px`,
          border: `${weight}px solid #fff`,
          borderColor: `${color} transparent transparent transparent`
        }}
      ></div>
      <div
        style={{
          width: `${size}px`,
          height: `${size}px`,
          border: `${weight}px solid #fff`,
          borderColor: `${color} transparent transparent transparent`
        }}
      ></div>
      <div
        style={{
          width: `${size}px`,
          height: `${size}px`,
          border: `${weight}px solid #fff`,
          borderColor: `${color} transparent transparent transparent`
        }}
      ></div>
      <div
        style={{
          width: `${size}px`,
          height: `${size}px`,
          border: `${weight}px solid #fff`,
          borderColor: `${color} transparent transparent transparent`
        }}
      ></div>
    </div>
  );
};

export default Loader;
