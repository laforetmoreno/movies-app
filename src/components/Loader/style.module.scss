.ldsRing {
  display: flex;
  position: relative;
  width: 64px;
  margin: 0 auto;
  justify-content: center;
  align-items: center;
}

.ldsRing div {
  box-sizing: border-box;
  display: block;
  position: absolute;
  margin: 6px;
  border-radius: 50%;
  animation: ldsRing 1.2s cubic-bezier(0.5, 0, 0.5, 1) infinite;
}

.ldsRing div:nth-child(1) {
  animation-delay: -0.45s;
}

.ldsRing div:nth-child(2) {
  animation-delay: -0.3s;
}

.ldsRing div:nth-child(3) {
  animation-delay: -0.15s;
}

@keyframes ldsRing {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}

.minHeight {
  min-height: 500px;
}
