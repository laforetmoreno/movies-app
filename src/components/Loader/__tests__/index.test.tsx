import React from "react";
import Link from "../Link.react";
import renderer from "react-test-renderer";

import Loader from "../index";

it("renders correctly", () => {
  const defaultProps = {
    size: 1,
    weight: 1,
    color: "",
    minHeight: true
  };

  const tree = renderer.create(<Loader {...defaultProps} />).toJSON();
  expect(tree).toMatchSnapshot();
});
