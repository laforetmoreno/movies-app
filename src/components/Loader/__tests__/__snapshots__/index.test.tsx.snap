// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="ldsRing minHeight"
>
  <div
    style={
      Object {
        "border": "1px solid #fff",
        "borderColor": " transparent transparent transparent",
        "height": "1px",
        "width": "1px",
      }
    }
  />
  <div
    style={
      Object {
        "border": "1px solid #fff",
        "borderColor": " transparent transparent transparent",
        "height": "1px",
        "width": "1px",
      }
    }
  />
  <div
    style={
      Object {
        "border": "1px solid #fff",
        "borderColor": " transparent transparent transparent",
        "height": "1px",
        "width": "1px",
      }
    }
  />
  <div
    style={
      Object {
        "border": "1px solid #fff",
        "borderColor": " transparent transparent transparent",
        "height": "1px",
        "width": "1px",
      }
    }
  />
</div>
`;
